### ISSUE TITLE
Training error

### ISSUE THREAD

**Issue by GreaBugs** [junjiehe96/FastInst#4]
Running train_ net.py file occurred Error:  TypeError :__ init__ () Got an unexpected keyword argument 'dtype', error location is detection2/engine/train_ loop.py
May I ask if anyone has encountered the same problem? How did you solve it?

**Comment by JIAZHAOQIAN** [comment]
This is because the API of pytorch has changed, torch.cuda.amp.autocast() does not have the  dtype parameter now. Please modify         "with autocast(dtype=self.precision)" to  "with autocast()" to fix this problem (line 491 in detectron2/detectron2/engine/train_loop.py).

**Comment by GreaBugs** [comment]
anaconda3/envs/fastinst/lib/python3.8/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection
  return lib.intersection(a, b, **kwargs)
**As you said, what is the reason for this error now?**

**Comment by JIAZHAOQIAN** [comment]
This is due to the update of setuptools. First, pip uninstall setuptools, then, pip or conda install setuptools==58.0.4.

**Comment by GreaBugs** [comment]
pip uninstall setuptools
pip install setuptools==58.0.4
                                                    and
pip uninstall shapely
pip install shapely==v1.8.5

**Comment by GreaBugs** [comment]
/data3/shenbaoyue/anaconda3/envs/fastry/lib/python3.8/site-packages/torch/nn/functional.py:718: **UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /opt/conda/conda-bld/pytorch_1623448278899/work/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)**

/data3/shenbaoyue/anaconda3/envs/fastry/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:129: **UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "**

May I ask how to solve these two types of errors during training? I tried to solve it according to the online operation, but it was not successful.
